src/Frontend/src/views/throughputreport/throughputClient.ts

```diff
@@ -34,9 +34,32 @@ class ThroughputClient {
     return data;
   }
 
-  public downloadReport(masks: string[]) {
-    //useDownloadFile();
-    //const response = await useFetchFromServiceControl(`${this.basePath}/report/file`);
+  public async downloadReport(spVersion: string) {
+    const response = await useFetchFromServiceControl(`${this.basePath}/report/file?${spVersion}`);
+    if (response.status == 200) {
+      var fileName = "throughput-report.json";
+      const contentDisposition = response.headers.get("Content-Disposition");
+      if (contentDisposition != null) {
+        try {
+          fileName = contentDisposition.split("filename=")[1].split(";")[0].replaceAll('"', "");
```

Comment:
I would use something like https://www.npmjs.com/package/@tinyhttp/content-disposition instead

Reply:
Our general view is to use fewer third party plugins so we have fewer dependencies that need updating. Is it worth introducing one for this?

Reply:
i think so, no need for us to implement this, given it is not straight forward

Reply:
> fewer third party plugins

btw, where is this documented?

Reply:
General guidance on third party dependencies [here](https://github.com/Particular/Platform/blob/main/principles/design-principles.md#third-party-dependencies).
A note on [keeping external components to a minimum in SP](https://github.com/Particular/Platform/blob/main/documentation/service-pulse.md#external-components). If we are adding one then there should be a reason and it needs to be added to this document.

Reply:
The first link seems to be very c# specific.
And the 2nd link doesn't really provide any reasons why!